packages/core/src/scope.ts

```diff
@@ -0,0 +1,126 @@
+import { Scope as BaseScope } from '@sentry/shim';
+import { Breadcrumb, SentryEvent, User } from '@sentry/types';
+
+/** An object to call setter functions on to enhance the event */
+export class Scope implements BaseScope {
+  /**
+   * Create a new empty internal scope. This will not be exposed to the user.
+   *
+   * @param breadcrumbs
+   * @param user
+   * @param tags
+   * @param extra
+   * @param fingerprint
+   * @param scopeChanged
+   */
+  public constructor(
+    public breadcrumbs: Breadcrumb[] = [],
+    public user: User = {},
+    public tags: { [key: string]: string } = {},
+    public extra: { [key: string]: any } = {},
+    public fingerprint?: string[],
+    private scopeChanged?: (scope: Scope) => void,
+  ) {}
+
+  /**
+   * Set internal on change listener.
+   */
+  public setOnChange(callback: (scope: Scope) => void): void {
```

Comment:
_nit_: For forward compatibility, it might be nice to have a signature like `.on(type: string, callback: (scope: Scope) => void)`